src/main/kotlin/com/atlassian/performance/tools/report/api/NonparametricFacade.kt

```diff
@@ -0,0 +1,58 @@
+package com.atlassian.performance.tools.report.api
+
+import com.numericalmethod.suanshu.stats.test.rank.wilcoxon.WilcoxonRankSum
```

Comment:
[Wilcoxon rank-sum test](https://en.wikipedia.org/wiki/Mann%E2%80%93Whitney_U_test) is an synonym for Mann–Whitney U test, which is [already represented in Apache commons-math](
http://commons.apache.org/proper/commons-math/javadocs/api-3.6.1/org/apache/commons/math3/stat/inference/MannWhitneyUTest.html), so we can get rid of the Suanshu snapshot.

Reply:
Unfortunately, `MannWhitneyUTest` from Apache commons-math does not allow to specify `mu` parameter that we use (which can be worked around), but what's more important, it does not have left-sided p-value calculation. :(

Reply:
[`mannWhitneyUTest` Javadoc](http://commons.apache.org/proper/commons-math/javadocs/api-3.6.1/org/apache/commons/math3/stat/inference/MannWhitneyUTest.html#mannWhitneyUTest(double[],%20double[])):
> Returns the asymptotic observed significance level, or p-value ...

Is that not it?

Reply:
No, it's a two-tailed p-value. But what we need is one-tailed p-value.
Instead of testing if the experiment **is the same** as the baseline, we will be testing if the experiment **is not worse** than the baseline.
Having only this requirement allows us to give more power to the test to detect the effect.

Reply:
This gave me some insight into the difference between one-tailed and two-tailed values: https://stats.idre.ucla.edu/other/mult-pkg/faq/general/faq-what-are-the-differences-between-one-tailed-and-two-tailed-tests/

So we're choosing the one-tailed (left) one, because we only care about detecting regressions and do not care about detecting improvements?

From the article above:
> Choosing a one-tailed test for the sole purpose of attaining significance is not appropriate.

This is worrying. Are you sure we're using it the right way?